src/games/strategy/triplea/ui/screen/IDrawable.java

```diff
@@ -180,6 +183,57 @@ public void draw(final Rectangle bounds, final GameData data, final Graphics2D g
     }
   }
 
+  /**
+   * Find the best rectangle inside the territory to place the name in. Finds the rectangle
+   * that can fit the name, that is the closest to the vertical center, and has a large width at
+   * that location. If there isn't any rectangles that can fit the name then default back to the
+   * bounding rectangle.
+   */
+  private Rectangle getBestTerritoryNameRect(final MapData mapData, final Territory territory,
+      final FontMetrics fontMetrics) {
+
+    // Find bounding rectangle and use to create parameters for a grid across the territory
+    final Rectangle territoryBounds = mapData.getBoundingRect(territory);
+    Rectangle result = territoryBounds;
+    final int maxX = territoryBounds.x + territoryBounds.width;
+    final int maxY = territoryBounds.y + territoryBounds.height;
+    final int centerY = territoryBounds.y + territoryBounds.height / 2;
+    final int minWidth = fontMetrics.stringWidth(territory.getName());
+    final int increment = fontMetrics.getAscent();
+    int maxScore = 0;
+
+    // Loop through the grid moving the starting point and determining max width at that point
+    for (int leftX = territoryBounds.x; leftX < maxX - increment; leftX += increment) {
+      for (int topY = territoryBounds.y; topY < maxY - increment; topY += increment) {
+        for (int rightX = maxX; rightX > leftX; rightX -= increment) {
+
+          // Find current rectangle and the 'score' based on how close to vertical center and width
+          final Rectangle current = new Rectangle(leftX, topY, rightX - leftX, increment);
+          final int verticalFromEdge = territoryBounds.height / 2 - Math.abs(centerY - (topY + increment));
```

Comment:
The `abs()` condition should just use different variables, or be written back in terms of territoryBounds so it can be understood more simply.